src/tty/unix.rs

```diff
@@ -865,7 +865,7 @@ impl Renderer for PosixRenderer {
             }
         }
         // we have to generate our own newline on line wrap
-        if end_pos.col == 0 && end_pos.row > 0 && !self.buffer.ends_with('\n') {
+        if end_pos.col == 0 && end_pos.row > 0 && !line.ends_with('\n') {
```

Comment:
We need to handle the case where there is an hint.

Reply:
Thank you for pointing out that. Do you have any suggestion? Another solution in my mind is to reverse search buffer, and skip escape sequence, take https://github.com/kkawakam/rustyline/blob/master/src/tty/mod.rs#L170 for reference.